internal/webhooks/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -184,9 +184,9 @@ func validateRAID(r *metal3api.RAIDConfig) []error {
 }
 
 func validateBMHName(bmhname string) error {
-	invalidname, _ := regexp.MatchString(`[^A-Za-z0-9\.\-\_]`, bmhname)
+	invalidname, _ := regexp.MatchString(`[^A-Za-z0-9\-]`, bmhname)
```

Comment:
There is more to validating a DNS name than this; e.g. it cannot start or end with a `-`, should be no more than 63 characters.

I suggest calling [this function](https://github.com/kubernetes/apimachinery/blob/f5dd29d6ada12819a4a6ddc97d5bdf812f8a1cad/pkg/util/validation/validation.go#L188-L203) to validate it.